### PR TITLE
Load compat in load-style.php

### PR DESCRIPTION
**Pull request by spacedmonkey**
Trac ticket:  https://core.trac.wordpress.org/ticket/58012

---
**This Pull Request is for code review only. Please keep all other discussion in the Trac ticket. Do not merge this Pull Request. See [GitHub Pull Requests for Code Review](https://make.wordpress.org/core/handbook/contribute/git/github-pull-requests-for-code-review/) in the Core Handbook for more details.**

### CODE REVIEW

**Comment by SergeyBiryukov** [comment]
Thanks! Yeah, that's what I was leaning towards in [58012.diff](https://core.trac.wordpress.org/attachment/ticket/58012/58012.diff) as well 🙂 

The `str_starts_with()` call was replaced in [r55710](https://core.trac.wordpress.org/changeset/55710), let's reinstate it in this PR too.

**Review by SergeyBiryukov — APPROVED** [review]

**Comment by azaozz** [comment]
Sorry but including that file is "Doing it wrong". `load-scripts.php` and `load-styles.php` are (were?) designed to run outside of WordPress. They use `script-loader.php` as a **whitelist of URLs to scripts and styles**. None of the functions there are supposed to run.

Looking at recent changes to `load-styles.php`, most `include ...` except `wp-admin/includes/noop.php` don't make sense and should be reverted.

File: src/wp-admin/load-styles.php [hunk 1]
```diff
@@ -16,6 +16,7 @@
 define( 'WP_CONTENT_DIR', ABSPATH . 'wp-content' );
 
 require ABSPATH . 'wp-admin/includes/noop.php';
+require ABSPATH . WPINC . '/compat.php';
```
Comment (azaozz) [inline src/wp-admin/load-styles.php:19]: Doesn't make sense to include a whole file just for this. `load-styles.php` is designed to run outside of WordPress.

**Review by azaozz — CHANGES_REQUESTED** [review]

**Review by johnbillion — CHANGES_REQUESTED** [review]
I agree with Andrew that this is an unnecessary change just to introduce what amounts to a coding standards change and has no functional benefit.

Loading the extra `compat.php` also presumably slows down the response time of these endpoints even further, albeit to a slight degree.

**Comment by spacedmonkey** [comment]
Closing as it is fixed.

**Pull request closed without merging**